### PR TITLE
Services, Network: introduction keepalive and refactoring

### PR DESCRIPTION
**Pull request by aarani**

### CODE REVIEW

File: NOnion/Network/TorCircuit.fs [hunk 1]
```diff
@@ -866,6 +866,9 @@ type TorCircuit
                     | RelayTruncated reason ->
                         let handleTruncated() =
                             match circuitState with
```
Comment (knocte) [inline NOnion/Network/TorCircuit.fs:868]: the 1st line of commit message is a title, therefore doesn't need a dot at the end

File: NOnion/Directory/NetworkStatusDocument.fs [hunk 2]
```diff
@@ -159,7 +161,12 @@ type RouterStatusEntry =
 
                     innerParse
                         { state with
-                            Flags = readRestAsString() |> Some
+                            Flags =
+                                readRestAsString()
+                                    .Split(
+                                        [| " " |],
```
Comment (knocte) [inline NOnion/Directory/NetworkStatusDocument.fs:167]: use Array.singleton

File: NOnion/Services/TorServiceClient.fs [hunk 3]
```diff
@@ -68,8 +68,8 @@ type TorServiceClient =
                 .Create()
                 .GetNonZeroBytes randomGeneratedCookie
 
-            let! endpoint, guardnode = directory.GetRouter false
-            let! _, rendNode = directory.GetRouter false
+            let! endpoint, guardnode = directory.GetRouter RouterType.Guard
+            let! _, rendNode = directory.GetRouter RouterType.Normal
```
Comment (knocte) [inline NOnion/Services/TorServiceClient.fs:71]: what does rendNode mean?

Comment (aarani) [reply NOnion/Services/TorServiceClient.fs:71]: rendezvous node

Comment (knocte) [reply NOnion/Services/TorServiceClient.fs:71]: please rename

File: NOnion/Directory/NetworkStatusDocument.fs (hunk 2 above)
Comment (aarani) [reply NOnion/Directory/NetworkStatusDocument.fs]: Done

File: NOnion/Network/TorCircuit.fs (hunk 1 above)
Comment (aarani) [reply NOnion/Network/TorCircuit.fs:868]: Removed

File: NOnion/NOnion.fsproj [hunk 4]
```diff
@@ -79,7 +79,6 @@
     <PackageReference Include="Fsharpx.Collections" Version="3.0.1" />
     <PackageReference Include="Portable.BouncyCastle" Version="1.8.10" />
     <PackageReference Include="System.Memory" Version="4.5.4" />
-    <PackageReference Include="System.Text.Json" Version="5.0.2" />
```
[the remark this answers is not in the document]
Comment (aarani) [reply NOnion/NOnion.fsproj:82]: Fixed

**Pull request merged**